features/frameworks/greentea-client/source/test_env.cpp

```diff
@@ -61,8 +61,10 @@ static void greentea_notify_version();
  * Rawserial object used to provide direct, raw serial communications
  * between the target and the host.
  */
-RawSerial greentea_serial(USBTX, USBRX); 
-
+RawSerial& greentea_serial() { 
+    static RawSerial serial(USBTX, USBRX);
```

Comment:
> The singleton pointer class is still good to create PODs without wrapping everything as a static, in a function and it is immune to the static initialization fiasco.

Hmm, I'm probably missing it, but I'm not sure I see where `SingletonPtr` benefits over static initialization other than the syntax difference, but both cases are different from the syntax for global objects.

Since these both solve the same problem of global-initialization, shouldn't we chose one solution and consolidate?